main.go

```diff
@@ -179,6 +182,17 @@ func main() {
 	renewDeadline := time.Second * time.Duration(leaderLeaseRenewDeadline)
 	retryPeriod := time.Second * time.Duration(leaderLeaseRetryPeriod)
 
+	// filter cache to subscribe to events from specific configmaps
+	newCache := cache.BuilderWithOptions(cache.Options{
+		SelectorsByObject: cache.SelectorsByObject{
+			&corev1.ConfigMap{}: {
+				Field: fields.AndSelectors(fields.SelectorFromSet(fields.Set{"metadata.name": "amazon-vpc-cni"}),
```

Comment:
need to go through this again tomorrow.

Reply:
Wondering how does this work?

```
// AndSelectors creates a selector that is the logical AND of all the given selectors
```

So does this mean we cache, ConfigMap with Name `amazon-vpc-cni` AND name `config.LeaderElectionKey` AND namesapce `kube-system`? This would result in nothing being watched I believe. Has this been tested?

Reply:
What we are looking for is OR or IN. But I don't think that's possible. [See](https://github.com/kubernetes/kubernetes/issues/32946)

Reply:
I've changed this to watch only for the amazon-vpc-cni configmap. 
Since we are no longer watching the leader election configmap, using the fieldSelector to match name and namespace of a single configmap should be good.